src/utils/DateUtils.ts

```diff
@@ -29,111 +130,174 @@ class DefaultDateToStringOptions extends Options implements IDateToStringOptions
   predefinedFormat: string = undefined;
 }
 
+/**
+ * This class modifies the received Date so that it is rendered the way the options specify.
+ */
 export class DateUtils {
-  static convertFromJsonDateIfNeeded(date: string): Date;
-  static convertFromJsonDateIfNeeded(date: number): Date;
-  static convertFromJsonDateIfNeeded(date: Date): Date;
+
   static convertFromJsonDateIfNeeded(date: any): Date {
+    return DateUtils.convertToStandardDate(date);
+  }
+
+  static convertToStandardDate(date: string): Date;
+  static convertToStandardDate(date: number): Date;
+  static convertToStandardDate(date: Date): Date;
+  /**
+   * Receives a date of any type and converts it to a standard Date object.
+   * @param date can be a string, a number or a Date object. It can also be any other type, but the result will be
+   * 'undefined' if it is not recognized as a date.
+   * @returns {any} the Date, or undefined it it wasn't recognized as a valid date.
+   */
+  static convertToStandardDate(date: any): Date {
```

Comment:
What purpose does it serve to have the first three methods with a typed argument if we will catch anything anyway with the argument types as `any`? Also is there a reason we fail silently if we can't parse the date? Could we throw an error?